test/realm_test.dart

```diff
@@ -552,5 +552,161 @@ Future<void> main([List<String>? args]) async {
       expect(() => realm.write(() => players[-1] = Person()), throws<RealmException>("Index out of range"));
       expect(() => realm.write(() => players[800] = Person()), throws<RealmException>());
     });
+
+    test('RealmList clear items from list', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create a team
+      final team = Team()..name = "Team";
+      realm.write(() => realm.add(team));
+
+      //Add players to the team
+      final newPlayers = [
+        Person()..name = "Michael Schumacher",
+        Person()..name = "Sebastian Vettel",
+        Person()..name = "Kimi Räikkönen",
+      ];
+      realm.write(() {
+        team.players.addAll(newPlayers);
+      });
+
+      //Ensure teams and players are in realm
+      var teams = realm.all<Team>();
+      expect(teams.length, 1);
+
+      var players = teams[0].players;
+      expect(players, isNotNull);
+      expect(players.length, 3);
+
+      //Clear list of team players
+      realm.write(() => teams[0].players.clear());
+
+      //Ensure that players are not reffered by the team in realm
+      teams = realm.all<Team>();
```

Comment:
we should not re-query the db here. I suggest we write the tests as any developer will write the. Also OSS tests are good documentation which someone might check how the API is supposed to be used.  I suggest we just remove this line here since the expectation should be that the  original team does not have players.

Reply:
Done.